ragmac_xdem/io.py

```diff
@@ -37,6 +37,65 @@ def dask_start_cluster(nproc, threads=1, ip_addres=None):
     print('Threads per worker:', threads, '\n')
     return client
 
+def optimize_dask_chunks(ds,
+                         min_chunk_size = 1,
+                         max_chunk_size = 10):
+    """
+    Function to optimize shape and size of dask chunk.
+    Finds square root of dim length, then increments chunk shape 
+    by square root until size > min_chunk_size and < max_chunk_size.
+    Chunk sizes measured in MiB.
+    """
+
+    ### More info at https://blog.dask.org/2021/11/02/choosing-dask-chunk-sizes#what-to-watch-for-on-the-dashboard
+    ### larger chunk sizes (e.g. 100 MiB) performed slower when analyzing the task graph and performance.
+    ### WIP
+    
+    t = len(ds.time)
+    x = len(ds.x)
+    y = len(ds.y)
+    
+    #Find something that is close to a multiple of dim length
```

Comment:
Technically finding fractals, not a multiples here.  Could find actual multiples < 100 with something like 

`multiple_of_n = [(i, int(n / i)) for i in range(1, int(n ** 0.5) + 1) if (n % i == 0)]` 

where `n` is the initial dim length, then take the largest multiple < 100 and use for slicing the array. Then use this as the smallest increment to step up the chunk dim length until a desired chunk size in MiB is reached.

Problem is that some numbers won't have multiples < 100 and greater than 1. In that case could just start with 100, but would end up with chunks at the end of each dim that aren't the same size as the rest. This imprecision in terms of finding actual multiples of the dim length comes out to be about the same as the currently implemented approach.

Either way, this tends to perform better than the automated chunking done with 

```
ds['band1'].data = ds['band1'].data.rechunk({0:-1, 1:'auto', 2:'auto'}, 
                                            block_size_limit=1e8, 
                                            balance=True)
```

What we need really need is to let dask do better mapping by either using `map_blocks` to distribute the function or `dask='allowed'` when using the built in xarray `ufunc` method.